### PR TITLE
Allow connection without hostname verification

### PR DESCRIPTION
**Pull request by sfackler**
cc #13

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -375,6 +375,21 @@ impl TlsConnector {
...
     }
+
+    /// Like `connect`, but does not validate the server's domain name against its certificate.
+    ///
+    /// # Warning
+    ///
+    /// You should think very carefully before you use this method. If hostname verification is not
+    /// used, *any* valid certificate for *any* site will be trusted for use from any other. This
+    /// introduces a significant vulnerability to man-in-the-middle attacks.
+    pub fn danger_connect_without_providing_domain_for_certificate_verification_and_server_name_indication<S>(
```
Comment (frewsxcv) [inline src/lib.rs]: 👌

**Comment by Philipp91** [comment]
If this is merged before https://github.com/sfackler/rust-native-tls/issues/3 is solved or another sensible way to use self-signed certificates (https://github.com/sfackler/rust-native-tls/issues/13#issuecomment-262103881) is available, people will start using `danger_connect_without_providing_domain_for_certificate_verification_and_server_name_indication` despite its name, because it's the only way.

That's of course their own risk. But during the time when this is the only solution, it might make its way onto platforms like StackOverflow as the "accepted solution", as was the case with Android here:
http://stackoverflow.com/a/7266768
Later, clear warnings were added to this answer, but they weren't present before. Research has shown that such code did in fact end up in production code of numerous Android apps (though it remains unclear if it was indeed copied from StackOverflow) and made them vulnerable.

Otherwise, I think the function name is well chosen and will attract enough attention.

**Comment by sfackler** [comment]
Self signed certificates will still fail to validate with this method - all that changes here is that we don't check the SAN/CN of the leaf.

**Comment by Philipp91** [comment]
Ah right.

**Pull request merged**